sdk/go/common/workspace/project.json

```
@@ -5,176 +5,290 @@
-            "description": "Author is an optional author that created this project.",
-            "type": ["string", "null"]
-        },
-        "website": {
-            "description": "Website is an optional website for additional info about this project.",
-            "type": ["string", "null"]
-        },
-        "license": {
-            "description": "License is the optional license governing this project's usage.",
-            "type": ["string", "null"]
-        },
-        "runtime": {
-            "title": "ProjectRuntimeInfo",
-            "oneOf": [
-                {
-                    "title": "Name",
-                    "type": "string",
-                    "minLength": 1
-                },
-                {
-                    "type": "object",
-                    "properties": {
-                        "name": {
-                            "title": "Name",
-                            "type": "string",
-                            "minLength": 1
-                        },
-                        "options": {
-                            "title": "Options",
-                            "type": "object",
-                            "additionalProperties": true
-                        }
-                    },
-                    "additionalProperties": false
-                }
-            ]
-        },
-        "main": {
-            "description": "Path to the Pulumi program. The default is the working directory.",
-            "type": ["string", "null"]
-        },
-        "config": {
-            "description": "Config directory relative to the location of Pulumi.yaml.",
-            "type": ["string", "null"],
-            "deprecated": true
-        },
-        "stackConfigDir": {
-            "description": "Config directory location relative to the location of Pulumi.yaml.",
-            "type": ["string", "null"]
-        },
-        "backend": {
-            "description": "Backend of the project.",
-            "type": ["object", "null"],
+          },
+          {
+            "type": "object",
             "properties": {
-                "url": {
-                    "description": "URL is optional field to explicitly set backend url",
-                    "type": "string"
-                }
+              "name": {
+                "title": "Name",
+                "type": "string",
+                "minLength": 1
+              },
+              "options": {
+                "title": "Options",
+                "type": "object",
+                "additionalProperties": true
+              }
             },
             "additionalProperties": false
+          }
+        ]
+      },
+      "main": {
+        "description": "Path to the Pulumi program. The default is the working directory.",
+        "type": [
+          "string",
+          "null"
+        ]
+      },
+      "config": {
+        "description": "A map of configuration keys to their types",
+        "type": "object",
```

```suggestion
        "type":  ["object", "string", "null"]
```
Still need to support the old string path here, and it's optional so a null is also valid.